plugins/topology/src/data-transforms/data-transformer.ts

```diff
@@ -33,6 +34,55 @@ export const getBaseTopologyDataModel = (resources: K8sResponseData): Model => {
     nodes: [],
     edges: [],
   };
+  VM_TYPES.forEach((key: string) => {
+    if (resources?.[key]?.data?.length) {
+      const typedDataModel: Model = {
+        nodes: [],
+        edges: [],
+      };
+      resources[key].data.forEach((resource: K8sWorkloadResource) => {
+        const item = createOverviewItemForType(key, resource);
+        if (item) {
+          const data = createTopologyNodeData(
+            resource,
+            item,
+            TYPE_VM,
+            'icon-default',
+            getUrlForResource(resources, resource),
+            {
+              podsData: getPodsDataForResource(resource, resources),
+              services: getServicesForResource(
+                resource,
+                resources.services?.data as V1Service[],
+              ),
+              ingressesData: getIngressesDataForResourceServices(
+                resources,
+                resource,
+              ),
+              routesData: getRoutesDataForResourceServices(resources, resource),
+            },
+          );
+
+          typedDataModel.nodes?.push(
+            getTopologyNodeItem(
+              resource,
+              TYPE_VM,
+              data,
+              WorkloadModelProps,
+              undefined,
+              undefined,
+              NodeShape.rect,
+            ),
+          );
+          mergeGroup(
+            getTopologyGroupItems(resource) as NodeModel,
+            typedDataModel.nodes as NodeModel[],
+          );
+        }
+      });
+      addToTopologyDataModel(typedDataModel, baseDataModel);
+    }
+  });
```

Comment:
Rather than adding an entire block for VM nodes, which is quite similar to the workload types block, you could simply check if the `key` is `virtualmachines`, evaluate the type and resources, and then push it to the topology model.